File: Server/src/main/java/core/game/content/dialogue/ZamboDialogue.java
package core.game.content.dialogue;

import core.game.node.entity.npc.NPC;
import core.plugin.Initializable;
import core.game.node.entity.player.Player;

/**
 * Represents the dialogue plugin used for zambo.
 * @author 'Vexia
 * @version 1.0
 */
@Initializable
public final class ZamboDialogue extends DialoguePlugin {

	/**
	 * Constructs a new {@code ZamboDialogue} {@code Object}.
	 */
	public ZamboDialogue() {
		/**
		 * empty.
		 */
	}

	/**
	 * Constructs a new {@code ZamboDialogue} {@code Object}.
	 * @param player the player.
	 */
	public ZamboDialogue(Player player) {
		super(player);
	}

	@Override
	public DialoguePlugin newInstance(Player player) {
		return new ZamboDialogue(player);
	}

	@Override
	public boolean open(Object... args) {
		npc = (NPC) args[0];
		interpreter.sendDialogues(npc, FacialExpression.HALF_GUILTY, "Hey, are you wanting to try some of my fine wines", "and spririts? All brewed locally on Karamja island.");
		stage = 0;
		return true;
	}

	@Override
	public boolean handle(int interfaceId, int buttonId) {
		switch (stage) {
		case 0:
			interpreter.sendOptions("Select an Option", "Yes, please.", "No, thank you");
			stage = 1;
			break;
		case 1:
			switch (buttonId) {
			case 1:
				end();
				npc.openShop(player);
				break;
			case 2:
				interpreter.sendDialogues(player, FacialExpression.HALF_GUILTY, "No, thank you.");
				stage = 20;
				break;
			}
			break;
		case 20:
			end();
			break;
		}
		return true;
	}

	@Override
	public int[] getIds() {
		return new int[] { 568 };
	}
}

File: Server/src/main/content/region/karamja/dialogue/ZamboDialogue.kt
package content.region.karamja.dialogue

import core.api.openNpcShop
import core.game.dialogue.DialoguePlugin
import core.game.dialogue.FacialExpression
import core.game.node.entity.npc.NPC
import core.game.node.entity.player.Player
import core.plugin.Initializable
import core.tools.END_DIALOGUE
import core.tools.START_DIALOGUE
import org.rs09.consts.NPCs

@Initializable
class ZamboDialogue(player: Player? = null) : DialoguePlugin(player) {

    override fun newInstance(player: Player?): DialoguePlugin {
        return ZamboDialogue(player)
    }

    override fun open(vararg args: Any?): Boolean {
        npc = args[0] as NPC
        npcl(
            FacialExpression.ASKING,
            " Hey, are you wanting to try some of my fine wines and spirits? All brewed locally on Karamja."
        ).also { stage = START_DIALOGUE }
        return true
    }

    override fun handle(interfaceId: Int, buttonId: Int): Boolean {
        when (stage) {
            0 -> options("Yes, please.", "No, thank you.").also { stage++ }

            1 -> when (buttonId) {
                1 -> playerl(
                    FacialExpression.FRIENDLY, "Yes, please."
                ).also { stage++ }

                2 -> playerl(
                    FacialExpression.FRIENDLY, "No, thank you."
                ).also { stage = END_DIALOGUE }
            }

            2 -> {
                openNpcShop(player, NPCs.ZAMBO_568)
                end()
            }
        }
        return true
    }

    override fun getIds(): IntArray {
        return intArrayOf(NPCs.ZAMBO_568)
    }
}